Feature/update sensor config

Blade-iii opened the pull request:
## This feature is the backend for the user story #13

### What's been added
- Database integration reads and updates the Configuration table based on user input.
- Config Factory to deal with reading the database for the Configuration table and populating a collection. 
- ~~Changes to database adding an method to update the database.~~ 
 
> [!NOTE]
> Whilst developing the feature i noticed Measurement frequency was not implemented in the models and not created in the database structure either so in a future pull request or commit this will be resolved.

Blade-iii committed:
Created a Factory method

Which can be used to expand upon the task in the future.

Blade-iii committed:
Added Database integration

Next step is to update the ui and start doing CRUD operations

Blade-iii committed:
Adding backend

Taking the entry from the user using Binding

Blade-iii committed:
Added update functionality to the database

Blade-iii committed:
Seperated the collections

Blade-iii committed:
Collection loading based on selection

This allows the user to view the content of the config table before making changes based on the dropdown selected

Blade-iii committed:
Added the rest of the sensor types

Created a method to handle string convertion to float because if you make the entry Bindings a float it will default the value to 0 which isn't very userfriendly

Blade-iii committed:
Database changes

Blade-iii committed:
Merge branch 'dev' into feature/update-sensor-config

Blade-iii committed:
Solved a merge conflict

And added scrollview

NikitaEdin commented on REA/ViewModels/UpdateSensorViewModel.cs:
```diff
@@ -0,0 +1,223 @@
+
+            foreach (var item in SelectedSensorCollection)
+            {
+                item.MinMeasurement = ConvertStringToFloat(sensorMaxValue);
+                item.MaxMeasurement = ConvertStringToFloat(sensorMaxValue);
```
There's a typo in the min and max measurement, as the max value is set for both variables.

It should be as follows:
```cs
item.MinMeasurement = ConvertStringToFloat(sensorMinValue);
item.MaxMeasurement = ConvertStringToFloat(sensorMaxValue);
```

NikitaEdin reviewed (changes requested):
Great work on implementing the `update sensor config` feature, which works great besides one thing which needs to be addressed before merging into dev.

The only noticeable issue is that `MinMeasurement` and `MaxMeasurement` values are being assigned with the same value in `UpdateConfig`, which shouldn't be the case.

> [!TIP]
>  Might consider adding input checks in the future or adding support for more types of sensors with less hardcoded checks.
> But nothing serious which affects this pull request from being approved.

Blade-iii committed:
Update UpdateSensorViewModel.cs

Fixed silly error caused by copy and pasting SensorMaxValue instead of typing it

Blade-iii commented:
@NikitaEdin Solved the typo thank you for spotting the mistake with the ```UpdateConfig``` method.
Also, thank you for the suggestions about the future of the feature as in it's current state if the administrator or other roles add more sensor types it will not be accounted for as the Sensor types are hard coded.
